Enterprise-CMCS/macpro-mdct-mcr
# [MDCT-2224, 2239] - Report Overlay Component and MLR Details Form

## Summary

This PR makes two large additions:
* A new component, called `EntityDetailsOverlay` which shows entity level form information using a new EntityProvider context.
* The MLR details page (using the EntityDetailsOverlay) and assorted form JSON.

### Description
Additionally, this PR adds a bunch of typings, which are separated out into their own commits. It also changes up the way that autosave works, making it aware of entity context (i.e "am I autosaving a field related to an entity") and correctly formatting the update request.

### Related ticket
[MDCT-2224](https://qmacbis.atlassian.net/browse/MDCT-2224)
[MDCT-2239](https://qmacbis.atlassian.net/browse/MDCT-2239)

### How to test
1. Create a new MLR report
2. Add a new plan
3. Select "Enter MLR"
4. Throw some values in the fields, and test autosave by tabbing away or unfocusing.
5. Verify that nested and un-nested choice fields (and their children) save correctly.

### Important updates
None, aside from some new typings that people should be aware of/use.

### Author checklist
